### PR TITLE
Invalid name, fix #3593

### LINKED ISSUE #3593: kdb ls `invalid key`: Bad error message
## Steps to Reproduce the Problem

Try either:

```sh
kdb ls user 
kdb ls user:
```

## Expected Result

That I get an error message like:

```
"user" is not a valid key name. A key name consists of [<namespace>:]/<path>.
```

## Actual Result

```
The command kdb ls terminated unsuccessfully with the info:
std::bad_alloc
Please report the issue at https://issues.libelektra.org/
```

## System Information

- Elektra Version: master

### PR DESCRIPTION
**Pull request by markus2330**
## Basics

These points need to be fulfilled for every PR:

- [x] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [x] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [x] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildservers are happy. If not, fix **in this order**:
  - [ ] add a line in `doc/news/_preparation_next_release.md`
  - [ ] reformat the code with `scripts/dev/reformat-all`
  - [ ] make all unit tests pass
  - [ ] fix all memleaks
- [x] The PR is rebased with current master.

If you have any troubles fulfilling these criteria, please write
about the trouble as comment in the PR. We will help you.
But we cannot accept PRs that do not fulfill the basics.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned every code not directly written by me in [THIRD-PARTY-LICENSES](doc/THIRD-PARTY-LICENSES)

## Review

Reviewers will usually check the following:

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

If you are already Elektra developer:

- Add the "work in progress" label if you do not want the PR to be reviewed yet.
- Add the "ready to merge" label **if the basics are fulfilled** and you also
  say that everything is ready to be merged.

### CODE REVIEW

**Comment by markus2330** [comment]
@mpranj would be good if this can also be part of 0.9.4 as the bug would be very confusing for beginners.

I did not check the formatting nor if all tests still succeed.. :-1:

File: src/bindings/swig/ruby/kdb.i [hunk 1]
```diff
@@ -123,7 +123,7 @@ the original C++ class in the following aspects:
 %catches (kdb::KeyTypeMismatch, kdb::KeyException) kdb::Key::getString;
 %catches (kdb::KeyTypeMismatch, kdb::KeyException) kdb::Key::getBinary;
 
-%catches (std::bad_alloc) kdb::Key::Key;
+%catches (std::KeyInvalidName) kdb::Key::Key;
```
Comment (mpranj) [inline src/bindings/swig/ruby/kdb.i:126]:
```suggestion
%catches (kdb::KeyInvalidName) kdb::Key::Key;
```

File: src/bindings/cpp/tests/testcpp_key.cpp [hunk 2]
```diff
@@ -545,7 +545,7 @@ TEST (key, valid)
 
 	for (auto & invalid_name : invalid_names)
 	{
-		EXPECT_THROW (Key (invalid_name, KEY_END), std::bad_alloc);
+		EXPECT_THROW (Key (invalid_name, KEY_END), std::KeyInvalidName);
```
Comment (mpranj) [inline src/bindings/cpp/tests/testcpp_key.cpp:548]:
```suggestion
		EXPECT_THROW (Key (invalid_name, KEY_END), kdb::KeyInvalidName);
```

**Review by mpranj — CHANGES_REQUESTED** [review]
Thank you so much for fixing it. You're right, this should be in 0.9.4.

I think there is only a namespace problem?

**Comment by markus2330** [comment]
Thank you so much for pointing this out! :sparkling_heart:

**Comment by markus2330** [comment]
One other commit: a266ebf3ac066aa219fd230e80eb1e7800d758e2 (hopefully the last&innocent)

**Pull request merged**